File: other/MT1051/week02/HW02/HW02_109601003.py
# -*- coding: utf-8 -*-
"""
Date: 2023/09/18
HW: 02
Author: 林群賀
Student Number: 109601003
"""

from typing import Any

class QuestionOne:
    bottom = 2.5
    height = 3.6

    def __call__(self, *args: Any, **kwds: Any) -> Any:
        pass

    def __init__(self) -> None:
        pass

    def get_bottom(self, ):
        return self.bottom
    
    def get_height(self, ):
        return self.height

    def get_area(self, ):
        area = self.get_bottom() * self.get_height() / 2

        return area

class QuestionTwo:
    radius = 2.8 / 2
    PI = 3.14159

    def __call__(self, *args: Any, **kwds: Any) -> Any:
        pass

    def __init__(self) -> None:
        pass

    def get_radius(self, ):
        return self.radius
    
    def get_PI(self, ):
        return self.PI
    
    def get_area(self, ):
        area = self.radius ** 2 * self.get_PI()

        return round(area, 2)

class QuestionThree:

    def __call__(self, *args: Any, **kwds: Any) -> Any:
        pass

    traingle_1_bottom = 1.6
    traingle_1_height = 2.2

    traingle_2_bottom = 2.6
    traingle_2_height = 3.2

    traingle_3_bottom = 3.6
    traingle_3_height = 4.5


    def __init__(self) -> None:
        pass

    def get_traingle_1_bottom(self,):
        return self.traingle_1_bottom
    
    def get_traingle_1_height(self,):
        return self.traingle_1_height

    def get_triangle_1_area(self,):
        triangle_1_area = (self.get_traingle_1_bottom() * self.get_traingle_1_height()) / 2

        return triangle_1_area
    
    def get_traingle_2_bottom(self,):
        return self.traingle_2_bottom
    
    def get_traingle_2_height(self,):
        return self.traingle_2_height
    
    def get_triangle_2_area(self,):
        triangle_2_area = (self.get_traingle_2_bottom() * self.get_traingle_2_height()) / 2

        return triangle_2_area
    
    def get_traingle_3_bottom(self,):
        return self.traingle_3_bottom
    
    def get_traingle_3_height(self,):
        return self.traingle_3_height
    
    def get_triangle_3_area(self,):
        triangle_3_area = (self.get_traingle_3_bottom() * self.get_traingle_3_height()) / 2

        return triangle_3_area
    
    def get_total_area(self, ):
        total_area = self.get_triangle_1_area() + self.get_triangle_2_area() + self.get_triangle_3_area()
        
        return round(total_area, 2)

class QuestionFour:

    external_radius = 3.2 / 2
    internal_radius = 1.5 / 2
    height = 6.8
    PI = 3.14159

    def __call__(self, *args: Any, **kwds: Any) -> Any:
        pass

    def __init__(self) -> None:
        pass

    def get_external_radius(self,):
        return self.external_radius
    
    def get_internal_radius(self,):
        return self.internal_radius
    
    def get_external_area(self,):
        external_area = self.external_radius ** 2

        return external_area
    
    def get_internal_area(self,):
        internal_area = self.internal_radius ** 2

        return internal_area
    
    def get_top_area(self,):
        top_area = self.get_external_area() - self.get_internal_area()
        
        return top_area
    
    def get_height(self,):
        return self.height
    
    def get_volume(self,):
        volume = (self.get_top_area() * self.PI * self.get_height())

        return round(volume, 2)
    
question_one = QuestionOne()
question_two = QuestionTwo()
question_three = QuestionThree()
question_four = QuestionFour()

area_one = question_one.get_area()
area_two = question_two.get_area()
area_three = question_three.get_total_area()
area_four = question_four.get_volume()

print("1.", area_one)
print("2.", area_two)
print("3.", area_three)
print("4.", area_four)

File: mt/mt1051/week02/HW02/HW02_109601003.go
// Date: 2023/09/18
// HW: 02
// Author: 林群賀
// Student Number: 109601003

package main

import (
	"fmt"
	"math"
)

type QuestionOne struct {
	Bottom float64
	Height float64
}

func (q *QuestionOne) GetArea() float64 {
	area := (q.Bottom * q.Height) / 2
	return area
}

type QuestionTwo struct {
	Radius float64
	PI     float64
}

func (q *QuestionTwo) GetArea() float64 {
	area := math.Pow(q.Radius, 2) * q.PI
	return math.Round(area*100) / 100
}

type QuestionThree struct {
	Triangle1Bottom float64
	Triangle1Height float64

	Triangle2Bottom float64
	Triangle2Height float64

	Triangle3Bottom float64
	Triangle3Height float64
}

func (q *QuestionThree) GetTriangleArea(bottom, height float64) float64 {
	area := (bottom * height) / 2
	return area
}

func (q *QuestionThree) GetTotalArea() float64 {
	area1 := q.GetTriangleArea(q.Triangle1Bottom, q.Triangle1Height)
	area2 := q.GetTriangleArea(q.Triangle2Bottom, q.Triangle2Height)
	area3 := q.GetTriangleArea(q.Triangle3Bottom, q.Triangle3Height)

	return math.Round((area1 + area2 + area3) * 100) / 100
}

type QuestionFour struct {
	ExternalRadius float64
	InternalRadius float64
	Height         float64
	PI             float64
}

func (q *QuestionFour) GetExternalArea() float64 {
	area := math.Pow(q.ExternalRadius, 2)
	return area
}

func (q *QuestionFour) GetInternalArea() float64 {
	area := math.Pow(q.InternalRadius, 2)
	return area
}

func (q *QuestionFour) GetTopArea() float64 {
	area := q.GetExternalArea() - q.GetInternalArea()
	return area
}

func (q *QuestionFour) GetVolume() float64 {
	volume := q.GetTopArea() * q.PI * q.Height
	return math.Round(volume*100) / 100
}

func main() {
	questionOne := QuestionOne{Bottom: 2.5, Height: 3.6}
	questionTwo := QuestionTwo{Radius: 2.8 / 2, PI: 3.14159}
	questionThree := QuestionThree{
		Triangle1Bottom: 1.6,
		Triangle1Height: 2.2,
		Triangle2Bottom: 2.6,
		Triangle2Height: 3.2,
		Triangle3Bottom: 3.6,
		Triangle3Height: 4.5,
	}
	questionFour := QuestionFour{
		ExternalRadius: 3.2 / 2,
		InternalRadius: 1.5 / 2,
		Height:         6.8,
		PI:             3.14159,
	}

	areaOne := questionOne.GetArea()
	areaTwo := questionTwo.GetArea()
	areaThree := questionThree.GetTotalArea()
	areaFour := questionFour.GetVolume()

	fmt.Println("1.", areaOne)
	fmt.Println("2.", areaTwo)
	fmt.Println("3.", areaThree)
	fmt.Println("4.", areaFour)
}
